pkg/stash/stasher.go

```diff
@@ -58,13 +58,13 @@ func (s *stasher) Stash(ctx context.Context, mod, ver string) (string, error) {
 	}
 	defer v.Zip.Close()
 	if v.Semver != ver {
-		exists, err := s.storage.Exists(ctx, mod, v.Semver)
-		if err != nil {
-			return "", errors.E(op, err)
-		}
-		if exists {
+		_, err = s.storage.Info(ctx, mod, v.Semver)
+		if err == nil {
 			return v.Semver, nil
 		}
+		if !errors.Is(err, errors.KindNotFound) {
+			return "", errors.E(op, err)
+		}
```

Comment:
@marwan-at-work can you remind me why we never had to check for other errors?

Reply:
@arschles there are no other errors, just not found and 500..and the not found is not really smart either thanks to `go mod download` not parsing VCS errors (which i understand is tricky)